pkg/exporters/k8sexporter/problemclient/problem_client.go

```diff
@@ -107,7 +108,7 @@ func (c *nodeProblemClient) SetConditions(newConditions []v1.NodeCondition) erro
 	if err != nil {
 		return err
 	}
-	return c.client.RESTClient().Patch(types.StrategicMergePatchType).Resource("nodes").Name(c.nodeName).SubResource("status").Body(patch).Do().Error()
+	return c.client.RESTClient().Patch(types.StrategicMergePatchType).Resource("nodes").Name(c.nodeName).SubResource("status").Body(patch).Do(context.TODO()).Error()
```

Comment:
I think we'd better to introduce context package into NPD, context.TODO() should not be used with product code.

Reply:
I agree, but let's make that a separate change. I think it's fine to use TODO for now.

Reply:
I've worked on a commit adding context last night, [here it is](https://github.com/kubernetes/node-problem-detector/pull/761/commits/e31f5a39613d34a8c9825803c7f14a0d9a4688b4) for reference.